### PR TITLE
Update storage codeowners

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Remove legacy owners

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -56,7 +56,7 @@
 /sdk/resourcemanager/         @ArcturusZhang @lirenhe @tadelesh
 
 # PRLabel: %Storage
-/sdk/storage/                 @mohsha-msft @zezha-msft @jhendrixMSFT
+/sdk/storage/                 @siminsavani-msft @souravgupta-msft @jhendrixMSFT
```
Comment (siminsavani-msft) [inline .github/CODEOWNERS:59]: Could we also add tasherif-msft here please?

**Review by siminsavani-msft — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (jhendrixMSFT) [reply .github/CODEOWNERS:59]: Yes, is there a preferred order?

Comment (siminsavani-msft) [reply .github/CODEOWNERS:59]: Nope, there is no preference!

Comment (jhendrixMSFT) [reply .github/CODEOWNERS]: done

**Review by siminsavani-msft — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**